src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -1216,6 +1217,76 @@ static void M1()
             await VerifyAnalyzerCSAsync(csSource);
         }
 
+        [Fact]
+        public async Task CallGuardFromPlatformSpecificAssembly()
+        {
+            string csDependencyCode = @"
+public class Library
+{
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""windows"")]
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""linux"")]
+    public static bool IsSupported => false;
+
+    public static void AMethod() { }
+}";
+            csDependencyCode = @$"[assembly: System.Runtime.Versioning.SupportedOSPlatform(""windows"")]
+                                  [assembly: System.Runtime.Versioning.SupportedOSPlatform(""linux"")]
```

Comment:
I see that the helper method below uses `"/PreviewAssembly/AssemblyInfo.g.cs"` to place the SupportedOSPlatform attributes for the assembly.

Would it make sense to have an extra test where the SupportedOSPlatform and the SupportedOSPlatformGuard attributes are within the same file?

What about a test case where the SupportedOSPlatform attribute is in a regular file (not a *.g.cs file)?

Reply:
> I see that the helper method below uses "/PreviewAssembly/AssemblyInfo.g.cs" to place the SupportedOSPlatform attributes for the assembly.

> Would it make sense to have an extra test where the SupportedOSPlatform and the SupportedOSPlatformGuard attributes are within the same file?

Well, there is plenty of tests, just all the other tests for this scenario are added within a same assembly, that is why this bug did not caught with them 

> What about a test case where the SupportedOSPlatform attribute is in a regular file (not a *.g.cs file)?

The attributes can be in any `*.cs` file of the assembly, it is just kind of convention that assembly level attributes added into `AssemblyInfo.cs`

Reply:
Thank you for the review, as suggested I added more scenarios like `UnsupportedOSPlatformGuard` attribute used for guard, the Guard property accessing internal field (which was causing a warning previously which fix resulted this bug).